### PR TITLE
+new double-click function

### PR DESCRIPTION
**Pull request by tolyanchenko**

### CODE REVIEW

File: test.js [hunk 1]
```diff
@@ -154,35 +150,217 @@ const render = () => {
       const newButton2 = document.createElement("button");
       newButton2.innerText = "x";
       newButton2.id = "newButton";
-      newButton2.addEventListener("click", cancelEditServerFinance);
-      newButton2.addEventListener("click", cancelEditServerFinance);
+      newButton2.addEventListener("click", cancelEditServer);
```
Comment (usatovaDasha) [inline test.js:153]: 2 identical listeners?

File: test.js [hunk 2]
```diff
@@ -154,35 +150,217 @@ const render = () => {
...
-      textDiv.className = "textDivClass";
-      const text = document.createElement("p");
-      text.id = index;
-      text.innerText = `${item.text} (${item.dateOfFinance})`;
+      if (index === editCash) {
+        const textDiv = document.createElement("div");
+        textDiv.className = "textDivClass";
+        const text = document.createElement("p");
+        text.id = index;
+        text.innerText = item.text;
```
Comment (usatovaDasha) [inline test.js:165]: destructurization item

File: test.js [hunk 3]
```diff
@@ -154,35 +150,217 @@ const render = () => {
...
+          text.id = index;
+          text.innerText = item.text;
+
+          textDiv.appendChild(text);
+          container.appendChild(textDiv);
+
+          const newInputDiv2 = document.createElement("div");
+          newInputDiv2.className = "newInputDiv";
+          const newInput2 = document.createElement("input");
+          newInput2.type = "text";
```
Comment (usatovaDasha) [inline test.js:219]: why not date?

File: test.js [hunk 4]
```diff
@@ -154,35 +150,217 @@ const render = () => {
...
       container.appendChild(newButtonDiv1);
+
+      content.appendChild(container);
     } else {
-      const textDiv = document.createElement("div");
-      textDiv.className = "textDivClass";
-      const text = document.createElement("p");
-      text.id = index;
-      text.innerText = `${item.text} (${item.dateOfFinance})`;
+      if (index === editCash) {
```
Comment (usatovaDasha) [inline test.js:160]: switch case

File: test.js [hunk 5]
```diff
@@ -154,35 +150,217 @@ const render = () => {
...
+            newButton1.addEventListener("click", stopEditServerFinance);
+            newButtonDiv.appendChild(newButton1);
+            container.appendChild(newButtonDiv);
+
+            const newButtonDiv1 = document.createElement("div");
+            newButtonDiv1.className = "newButtonDiv";
+            const newButton2 = document.createElement("button");
+            newButton2.innerText = "x";
+            newButton2.id = "newButton";
+            newButton2.addEventListener("click", cancelEditServer);
```
Comment (usatovaDasha) [inline test.js:307]: 2 identical listeners?

File: test.js [hunk 6]
```diff
@@ -154,35 +150,217 @@ const render = () => {
...
+          newButton1.addEventListener("click", stopEditServerFinance);
+          newButtonDiv.appendChild(newButton1);
+          container.appendChild(newButtonDiv);
+
+          const newButtonDiv1 = document.createElement("div");
+          newButtonDiv1.className = "newButtonDiv";
+          const newButton2 = document.createElement("button");
+          newButton2.innerText = "x";
+          newButton2.id = "newButton";
+          newButton2.addEventListener("click", cancelEditServer);
```
Comment (usatovaDasha) [inline test.js:249]: 2 identical listeners??

File: test.js [hunk 7]
```diff
@@ -154,35 +150,217 @@ const render = () => {
...
+        newButton1.addEventListener("click", stopEditServerFinance);
+        newButtonDiv.appendChild(newButton1);
+        container.appendChild(newButtonDiv);
+
+        const newButtonDiv1 = document.createElement("div");
+        newButtonDiv1.className = "newButtonDiv";
+        const newButton2 = document.createElement("button");
+        newButton2.innerText = "x";
+        newButton2.id = "newButton";
+        newButton2.addEventListener("click", cancelEditServer);
```
Comment (usatovaDasha) [inline test.js:199]: 2 identical listeners?

File: test.js [hunk 8]
```diff
@@ -154,35 +150,217 @@ const render = () => {
...
+            text.id = index;
+            text.innerText = item.text;
+
+            const dateDiv = document.createElement("div");
+            dateDiv.className = "dateDivClass";
+            const dateText = document.createElement("p");
+            dateText.id = index;
+            dateText.innerText = `(${item.dateOfFinance})`;
+
+            text.ondblclick = function () {
```
Comment (usatovaDasha) [inline test.js:324]: es6

File: test.js [hunk 9]
```diff
@@ -154,35 +150,217 @@ const render = () => {
...
+              editText = index;
+              editFinance = index;
+              render();
+            };
+            dateText.ondblclick = function () {
```
Comment (usatovaDasha) [inline test.js:329]: es6

File: test.js [hunk 10]
```diff
@@ -154,35 +150,217 @@ const render = () => {
...
+            text.innerText = item.text;
+
+            const dateDiv = document.createElement("div");
+            dateDiv.className = "dateDivClass";
+            const dateText = document.createElement("p");
+            dateText.id = index;
+            dateText.innerText = `(${item.dateOfFinance})`;
+
+            text.ondblclick = function () {
+              editText = index;
```
Comment (usatovaDasha) [inline test.js:325]: why you need save  editText and editFinance?

File: test.js [hunk 11]
```diff
@@ -154,35 +150,217 @@ const render = () => {
...
+              editFinance = index;
+              render();
+            };
+            dateText.ondblclick = function () {
+              editDate = index;
```
Comment (usatovaDasha) [inline test.js:330]: higher

File: test.js [hunk 12]
```diff
@@ -154,35 +150,217 @@ const render = () => {
...
+            textDiv.appendChild(dateText);
+            container.appendChild(textDiv);
+
+            const cashDiv = document.createElement("div");
+            const cash = document.createElement("p");
+            cash.innerText = item.cash;
+            cashDiv.className = "cashDivClass";
+            cashDiv.id = "cashDivId";
+
+            cash.ondblclick = function () {
```
Comment (usatovaDasha) [inline test.js:345]: es6

File: test.js [hunk 13]
```diff
@@ -154,35 +150,217 @@ const render = () => {
...
+              editCash = index;
```
Comment (usatovaDasha) [inline test.js:346]: higher

File: test.js (hunk 2 above)
Comment (tolyanchenko) [reply test.js:165]: can't be done. "text" is already declared at 163.

File: test.js (hunk 10 above)
Comment (tolyanchenko) [reply test.js:325]: editText, editDate and editCash must be there so we could create inputs and change info.  
editFinance is there to change all info at once.

**Pull request merged**